### PR TITLE
NEW Update RelatedPages feature for ManyManyThrough

### PR DESCRIPTION
**Pull request by NightJar**
Which also provides the ability to version the relationship.

relies on https://github.com/symbiote/silverstripe-gridfieldextensions/pull/260

### CODE REVIEW

File: src/Model/RelatedPageLink.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    private static $extensions = [
+        Versioned::class
+    ];
+
+    private static $db = [
+        'SortOrder' => DBInt::class
+    ];
+
+    // For backwards compatibility these must match a traditional 'many_many' definition.
```
Comment (robbieaverill) [inline src/Model/RelatedPageLink.php]: Please use `/** */` comment syntax for PHPDocs

**Comment by NightJar** [comment]
OK, so I've refactored a little bit to preserve the existing `many_many` relationship that will still return the expected `ManyManyList`, while maintaining the many many _through_ under a new name, and using that to preform the ordering. This way both are available, for a very minimal overhead in technical debt (50 consecutive characters).

Since both relationships use the same backing data, hopefully this will see Backwards Compatibility truly maintained.

**Comment by robbieaverill** [comment]
This PR requires https://github.com/symbiote/silverstripe-gridfieldextensions/pull/260 right? If so, should we target it at 2.2 at an enhancement

**Comment by NightJar** [comment]
In regards to this module, I'd regard it as a bugfix. It requires a fix from a dependency, but I think the constraint already allows for it.

Lemme check.

**Comment by dhensby** [comment]
Composer dependencies don't resolve on travis :(

**Comment by NightJar** [comment]
I'm aware; should hopefully be a simple matter to resolve - it's more finding the time for a context switch that's preventing this happening quickly.

**Comment by NightJar** [comment]
Thank you @robbieaverill ❤️

**Pull request merged**